pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -339,6 +329,30 @@ func updateStatusFromBuildStatus(taskRun *v1alpha1.TaskRun, buildStatus buildv1a
 			ContainerState: *state.DeepCopy(),
 		})
 	}
+
+	if bs := buildStatus.GetCondition(duckv1alpha1.ConditionSucceeded); bs != nil {
+		// Build Failure
+		if bs.IsFalse() {
+			// if Retry happnens, don't do anything else
+			taskRunStatus := taskRun.Status.DeepCopy()
+			taskRunStatus.Conditions = buildStatus.Conditions
+			if retry(taskRun, taskRunStatus) {
+				//return
```

Comment:
why commented? Irrespective of `return` code path remains the same but over time that might change so I would recommend uncommenting `return` statement.